### DISCUSSION TITLE
2023 Sole E35 compatibility

### DISCUSSION THREAD

**Discussion by realericmc** [cagnulein/qdomyos-zwift#2195] [Q&A]
I see Sole E25 in the compatibility list, but has anyone used a 2023 Sole E35 elliptical?  It looks like the SW in those is fairly different than the older models.

**Comment by cagnulein** [comment]
hi @realericmc send here a nrfconnect screenshot about your elli and i will check it. nrfconnect is a free app on the store

let me know!

**Reply by realericmc** [reply]
Hi @cagnulein, I bought one and tried it.  In the qdomyos app it showed up, but there was no data.  In nrf connect it shows up as a fitness device, but is not connecting.  Any suggestions on tricks to get more out of it?  thanks!
![IMG_0843](https://github.com/cagnulein/qdomyos-zwift/assets/41210335/3b135c74-a4ca-4706-a34b-1012aa6c783b)

**Reply by cagnulein** [reply]
@realericmc Perfect could you also please send me the client tab screenshot? All the tab long

Thanks

**Reply by realericmc** [reply]
okay, it didn't work on mone, but my iPad got it.  thanks!
![IMG_0215](https://github.com/cagnulein/qdomyos-zwift/assets/41210335/b61404cb-6352-41cc-b1a2-5ad54389dd7a)

![IMG_0217](https://github.com/cagnulein/qdomyos-zwift/assets/41210335/8f2a635c-21aa-4ca0-8c6f-5c1e8185d6cd)
y ph
![IMG_0218](https://github.com/cagnulein/qdomyos-zwift/assets/41210335/5963390a-da76-43a8-8ef1-8fdf17439255)
![IMG_0216](https://github.com/cagnulein/qdomyos-zwift/assets/41210335/81da5571-2448-4537-a1f3-19a749f1ab79)

**Reply by cagnulein** [reply]
perfect! I'm adding it! Did you have already the QZ app installed on your phone?

**Reply by realericmc** [reply]
cool!  thanks!  yep, I have the app.  no problem switching to a TestFlight version though if you need

**Reply by cagnulein** [reply]
perfect! 
send me an email to roberto.viola83 at gmail.com indicating this ticket! Thanks

**Reply by realericmc** [reply]
on version 771-
-cadence is reading, but looks to be double.  What was sent to swift was double that (so 4x the rpm)
-speed is staying zero
-odometer is updating at a bit less than 2x what is shown on elliptical
-resistance is shown in the app and is updating when changed on the elliptical.  
The buttons on the tile are not changing it (it briefly changes, then goes back to what the elliptical already had)
-inclination is staying zero
-HR is showing 0-4 and I can’t correlate it to actual HR
-fan speed is not updating when the fan is turned on (no biggie, just for completeness)
-power is staying zero

[debug-Thu_Mar_21_13_03_59_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14705360/debug-Thu_Mar_21_13_03_59_2024.log)

**Reply by cagnulein** [reply]
@realericmc ok i'm about to send you a new version. What I did:
- cadence fixed, on zwift there is a setting under accessories, power sensor setting, "Doubling Cadence for Run", enable it if you want the half on zwift.
- speed fixed hopefully
- odometer fixed hopefully
- for the buttons, i didn't see in the log. can you please do it again?
- inclination fixed hopefully
- for the hr, in the heart rate setting enable the "disable hr from the machinery"
- power fixed hopefully

let me know!

**Reply by realericmc** [reply]
Thanks Roberto!

Sorry for the slow response (different timezone).  Thanks for the new version.  Going through your list:

- cadence fixed, on zwift there is a setting under accessories, power sensor setting, "Doubling Cadence for Run", enable it if you want the half on zwift.
   - Cadence seems to be using a count of rotations now instead of a rate.  It is continually increasing
- speed fixed hopefully
   - Looks good
- odometer fixed hopefully
   - Looks good
- for the buttons, i didn't see in the log. can you please do it again?
   - Attached
- inclination fixed hopefully
   - Stays zero and the buttons in the app didn’t have an effect
   - The resistance now also stays zero
- for the hr, in the heart rate setting enable the "disable hr from the machinery”
   - I’m fine with skipping this.  I think my hr strap can pair to multiple things
- power fixed hopefully
   - Shows 256 when there is power applied, zero when not

[debug-Fri_Mar_22_08_09_24_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14723311/debug-Fri_Mar_22_08_09_24_2024.log)

**Reply by cagnulein** [reply]
thanks, i will check it ASAP, in the weekend i'm trying to stay with my family otherwise I'm risking a divorce :)

**Reply by cagnulein** [reply]
last version before the weekend.

- i should have fixed he wattage and the cadence issue
- inclination should works too hopefully
- i tried to add the command for the resistance, let's see if it will work

**Reply by realericmc** [reply]
Thanks!

- Wattage and cadence look good
- inclination shows 10x what the elliptical shows, which is a number from 1-20.  In terms of the inclination of the rails, that is 2º to 15º according to my phone magnetometer 
- resistance is updating correctly now in the app when I do it at the elliptical.  the buttons on the app for both resistance and inclination still have no effect

**Reply by cagnulein** [reply]
@realericmc i fixed the inclination thing (i will send it later). For the resistance buttons: are you able to change the resistance from the sole app already? I have an idea in case. Let me know!

**Reply by realericmc** [reply]
Hi @cagnulein .  yep, I just tried the sole+ app via bluetooth and it can change both the inclination and resistance.

btw, it would be nice if the heart rate could be passed through.  I tried this with my setup with zwift this weekend and was one bt connection short without that.  thanks!

**Reply by cagnulein** [reply]
ok perfect, let me try to add both resistance and inclination so. for the heart rate, do you mean using an Heart Rate device connected to QZ or using the HR from the sole device itself?

**Reply by realericmc** [reply]
cool, thanks!  

I meant from the device if possible.  The elliptical can make some adjustments based on hr so I would like to have it available there.  I tried to pair with the app and pick that up at the elliptical, but it wouldn't pair.  It did work pairing the app with zwift, but that didn't help since that was the same number of connections to my strap and I couldn't connect it also to the elliptical and also my bt fan.

**Reply by cagnulein** [reply]
@realericmc ok could you please provide me a debug log with the heart rate attached and tell me which values do you read in the sole device directly? also, qz, has a built in hr feature to adjust the resistance and the inclination itself following an heart rate zone! also qz can handle the bt fan too, changing the speed dinamically using the hr or the wattage or the speed!

**Reply by realericmc** [reply]
Thanks @cagnulein !

Here is a log.  the hr should be in the 70s and 80s.  Poking around through the log, the most likely suspect to me is the third byte from the end in the packets.  For example, the 4a in this one:
Mon Mar 25 13:30:04 2024 1711391404711 Debug: /Users/cagnulein/qdomyos-zwift/src/devices/bluetooth.cpp void bluetooth::debug(const QString &) " << fc 0b 00 00 00 00 30 00 14 00 00 01 00 00 00 05 00 02 00 00 00 00 4a 32 00"

Nice to have those features!  for me it would still be more convenient though to direct connect with my fan, since I use that same fan with my rower in that fashion.  To do it with the app, I would have to repair every time I switch machines since the kickr headwind I have only remembers the last pairing.

[debug-Mon_Mar_25_13_28_55_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14748108/debug-Mon_Mar_25_13_28_55_2024.log)

**Reply by cagnulein** [reply]
ok sent a new one. heart rate and inclination should be right now. also i added the possibily to force the inclination and resistance, let's see if they work, otherwise send me a fresh debug log. thanks!

**Reply by realericmc** [reply]
Thanks @cagnulein!

HR is working, inclination also looks good.  I tried the app buttons for resistance and inclination but still no effect.

[debug-Tue_Mar_26_08_17_41_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14758975/debug-Tue_Mar_26_08_17_41_2024.log)

**Reply by cagnulein** [reply]
ok thanks @realericmc i guess i got the reason why the resistance/inclination can't be controlled. I'm sending you a new one, hopefully the last one :D

**Reply by cagnulein** [reply]
Sent!

**Reply by realericmc** [reply]
Thanks @cagnulein!

well....  the buttons still didn't work.  wouldn't it be nice if there were standards for this stuff :-)

 
[debug-Tue_Mar_26_11_43_53_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14761858/debug-Tue_Mar_26_11_43_53_2024.log)

**Reply by cagnulein** [reply]
@realericmc don't test the 776 build, i'm building a new one instead

**Reply by cagnulein** [reply]
@realericmc just sent the 777. go with this :)

**Reply by realericmc** [reply]
thanks @cagnulein!

easy to skip versions when they come that early :-)

so close....   resistance works.  incline works to decrease, but not to increase

**Reply by cagnulein** [reply]
great! create me a debug log and i will fix this too!

**Reply by realericmc** [reply]
ah, sorry.  coffee hadn't kicked in yet.  here you go


[debug-Wed_Mar_27_08_24_17_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14774764/debug-Wed_Mar_27_08_24_17_2024.log)

**Reply by cagnulein** [reply]
@realericmc the log is corrupted, you have to wait the upload before clicking on reply. thanks

**Reply by realericmc** [reply]
sorry about that.  I fixed it in that message

**Reply by cagnulein** [reply]
@realericmc does your elliptical has a 0.5 step for inclination or only 1% grade?

**Reply by realericmc** [reply]
It has 20 "levels" and I have not seen any indication that it can increment in half steps.  The levels are not equivalent to percent grade though.  I couldn't find that in their specs, but I measured it at 2º for level 1 through 15º for level 20.

**Reply by cagnulein** [reply]
perfect! that's the issue so! set the "Inclination Step:" under the treadmill setting to 1 and i guess it will work on both direction! i'm also doing this in the code for this particular model. Let me know!

**Reply by realericmc** [reply]
yep, nailed it.  Changing that to 1 worked.  Thanks!

**Reply by cagnulein** [reply]
amazing!

**Reply by jncanches** [reply]
@realericmc may I ask you what E35 you have ? Is it the one with 10.1' touch display (with apps pre-installed) or the one with a non-touch matrix LCD display? Does yours have FTMS bluetooth 5 ?
I have the non-FTMS bluetooth 4 e35 and I cannot control inclination or resistance.
Thank you :-)

**Comment by realericmc** [comment]
Thanks @cagnulein!  I am in the shopping phase so don't have one yet.  I will try to find one in a store and get you the screenshot.

**Comment by cagnulein** [comment]
Thanks I will check it tomorrow

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 21 mar 2024 alle 19:12 Eric McMurry ***@***.***>
ha scritto:

**Comment by realericmc** [comment]
Thanks!

Sent some coffee to drink while you look at it.  Please let me know if there is something else you would like me to try.

**Comment by realericmc** [comment]
That is an easy call :-)

Have a great weekend!

**Comment by realericmc** [comment]
Thanks Roberto!

I’ll give that a go when the link comes by.  Have a great weekend!

**Comment by cagnulein** [comment]
Sent!
Have a great weekend too!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 22 mar 2024 alle ore 16:05 Eric McMurry <
***@***.***> ha scritto:

> Thanks Roberto!
>
> I’ll give that a go when the link comes by. Have a great weekend!
>

**Comment by cagnulein** [comment]
perfect i will check hr tomorrow. why don't use the fan and the rower too?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 25 mar 2024 alle 19:47 Eric McMurry ***@***.***>
ha scritto:

**Reply by realericmc** [reply]
thanks!  for the rower I pair the hr strap with it directly (ant), and then pair the rower directly to the exr app.  I then use the bt pairing from my hr strap with the fan and run that in hr mode.  That works well.  If I were to use the qz app with the fan when I am on the elliptical though, I would have to re pair the fan each time I switch between machines, which can be in the same workout.

btw, on the pairing topic, I have been having to do repairing and restarting multiple times each time I come back to the elliptical to get it working with the qz app.  I have not figured out yet what combination of things works best yet, but it has consistently not worked after the elliptical has gone to sleep without a minimum of forgetting the pairing.

**Comment by cagnulein** [comment]
i mean if you use the fan with qz you can connect the fan directly on qz
all the time without repairing it each time.

for the pairing issue just use the manual device in the advanced setting to
set it to your sole device, for sure qz is connecting to the rower too

you can use different profiles in the app to connect to one or the other

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 25 mar 2024 alle 19:57 Eric McMurry ***@***.***>
ha scritto:

**Reply by realericmc** [reply]
yep, that would work.  If I am just rowing I don't bring my phone though.

I did try the manual device in the advance setting, but I am still having to do multiple restarts and repairings.  Perhaps it is an issue with the elliptical

Thanks!

**Comment by jncanches** [comment]
> I see Sole E25 in the compatibility list, but has anyone used a 2023 Sole E35 elliptical?  It looks like the SW in those is fairly different than the older models.

@realericmc may I ask you what E35 you have ? Is it the one with 10.1' touch display (with apps pre-installed) or the one with a non-touch matrix LCD display? Does yours have FTMS bluetooth 5 ?
I have the non-FTMS bluetooth 4 e35 and I cannot control inclination or resistance.
Thank you :-)

**Reply by realericmc** [reply]
I have the one with the 10" touch screen and the apps.  I'm not sure about FMTS support, but it does support inclination and resistance control after Roberto did some work for it.

**Reply by cagnulein** [reply]
@jncanches Are you able to control inclination and resistance from the manufacturer app? If so i can add it as well

**Reply by realericmc** [reply]
sorry, I wasn't clear.  It already works in qdomyos (at least with the test versions you sent me last year, I don't recall if I had tested a released version).  If @jncanches has the same kind of E35 with the touch screen as I do, it should work fine now.

**Reply by jncanches** [reply]
Sorry for this answer delay. After I asked this question, I realized my e35 model is not a 2022 model, but a 2019 one. It cannot be controlled, even with Sole+ app.
Thank you for this project. I asked for an exchange with a 2024 version. As soon as I receive the new one, I will test qdomyos :)
